R/fit.R

```diff
@@ -599,10 +596,10 @@ CmdStanFit$set("public", name = "output", value = output)
 #' }
 #'
 metadata <- function() {
-  if (!length(self$output_files(include_failed = FALSE))) {
-    stop("Fitting failed. Unable to retrieve the metadata.", call. = FALSE)
-  }
   if (is.null(private$metadata_)) {
+    if (!length(self$output_files(include_failed = FALSE))) {
+      stop("Fitting failed. Unable to retrieve the metadata.", call. = FALSE)
+    }
```

Comment:
Move this error after the `is.null` check so that it's only thrown if metadata hasn't already been read in.